Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -58,7 +62,14 @@ public enum MacroDefinition {
   /// defining macro. These subtrees will need to be replaced with the text of
   /// the corresponding argument to the macro, which can be accomplished with
   /// `MacroDeclSyntax.expandDefinition`.
-  case expansion(MacroExpansionExprSyntax, replacements: [Replacement])
+  case expansion(MacroExpansionExprSyntax, replacements: [Replacement], genericReplacements: [GenericArgumentReplacement])
+}
+
+extension MacroDefinition {
+  /// Best effort compatibility shim, the case has gained additional parameters.
+  public func expansion(_ node: MacroExpansionExprSyntax, replacements: [Replacement]) -> Self {
```

Comment:
Let’s add a deprecation message to the compatibility shim.

```suggestion
  @available(*, deprecated, message: "Use the expansion case with three associated values instead")
  public func expansion(_ node: MacroExpansionExprSyntax, replacements: [Replacement]) -> Self {
```